### PR TITLE
read config map and add labels to namespaces

### PR DESCRIPTION
**Pull request by gearama**
moved the unit tests to mocha, 
added config reader 
added namespace label
all in one night

### CODE REVIEW

File: src/ConfigReader.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                    const parsedContent = toml.parse(fileContent)
+                    config = new AddonConfig(parsedContent);
+                } else {
+                    logger.info(`config does not exist, using defaults`);
+                }
+                logger.info(`excluded namespaces ${config.excludedNamespaces}`);
+                resolve(config);
+            })
+        }).catch(err => {
+            logger.error(`${err}`);
```
Comment (tokaplan) [inline src/ConfigReader.ts:30]: Will we see this output anywhere?

File: src/NamespaceLabeler.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            this.delay = delay*60*1000;
+        }
+
+        setTimeout(async () => {
+            logger.info("relooping")
+            await this.labelNamespace();
+        }, this.delay)
+    }
+
+    private async labelNamespace():Promise<any> {
```
Comment (tokaplan) [inline src/NamespaceLabeler.ts:21]: When is this invoked?

File: src/NamespaceLabeler.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    logger.info(`got namespace list ${JSON.stringify(result)}`);
+                    const namespaceList = result.body.items;
+                    namespaceList.forEach((item: V1Namespace) => {
+                        const patchPayload = item;
+                        if (patchPayload.metadata.labels == null) {
+                            patchPayload.metadata.labels = {};
+                        }
+                        patchPayload.metadata.labels["app-monitoring"] =
+                            config.excludedNamespaces.indexOf(item.metadata.name) < 0 ?
+                            "enable" : "disabled";
```
Comment (tokaplan) [inline src/NamespaceLabeler.ts:39]: Does this mean we're adding labels app-monitoring=false on namespaces that we don't care about? Is that the intention? Can we just not add anything for those namespaces or remove the label for them?

File: src/ConfigReader.ts (hunk 1 above)
Comment (gearama) [reply src/ConfigReader.ts:30]: yes you see it at the console for the moment, just like we see all the other logs in the pod

File: src/NamespaceLabeler.ts (hunk 2 above)
Comment (gearama) [reply src/NamespaceLabeler.ts:21]: in the method above, the loop method

File: src/NamespaceLabeler.ts (hunk 3 above)
Comment (gearama) [reply src/NamespaceLabeler.ts:39]: yeah we can , i was more looking to see the result when executing it, i'll fix it

**Pull request merged**